code/src/pages/MovieDetail.js

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from "react";
+import { Link, useParams } from "react-router-dom";
+
+import { ErrorPage } from "./ErrorPage";
+import { Loading } from "../components/Loading";
+
+export const MovieDetail = () => {
+  const { movieId } = useParams();
+  let DETAIL_URL = `https://api.themoviedb.org/3/movie/${movieId}?api_key=f8d93b3169765fb54ceb97a483ec4821&language=en-US`;
+  const [movie, setMovie] = useState({});
+  const [errorPage, setErrorPage] = useState(false);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    fetch(DETAIL_URL)
+      .then((response) => response.json())
+      .then((json) => setMovie(json));
+  }, [DETAIL_URL]);
+
+  useEffect(() => {
+    if (movie.success === false) {
+      setErrorPage(true);
+    }
+    if (movie) {
+      setLoading(false);
+    }
+  }, [movie, errorPage, loading]);
```

Comment:
❗❗ This structure is causing the fetch to fail at first and then succeed. Cause it calls it twice (I think). I had the same issue in my project and solved it by moving setLoading to the fetch function.
This is the error I get: 
![image](https://user-images.githubusercontent.com/3659568/114057705-15e6e500-9893-11eb-99f8-faff802a1470.png)

**Here is my solution**

```suggestion
  useEffect(() => {
    setLoading(true);
    fetch(DETAIL_URL)
      .then((response) => response.json())
      .then((json) => {
        setMovie(json)
        setLoading(false);
      });
  }, [DETAIL_URL]);

  useEffect(() => {
    if (movie.success === false) {
      setErrorPage(true);
    }
  }, [movie, errorPage]);
```

Reply:
Done, it worked